ast35/Grammar/Grammar

```diff
@@ -30,9 +30,10 @@ async_funcdef: ASYNC funcdef
 funcdef: 'def' NAME parameters ['->' test] ':' [TYPE_COMMENT] suite
 
 parameters: '(' [typedargslist] ')'
-typedargslist: (tfpdef ['=' test] (',' tfpdef ['=' test])* [','
-       ['*' [tfpdef] (',' tfpdef ['=' test])* [',' '**' tfpdef] | '**' tfpdef]]
-     |  '*' [tfpdef] (',' tfpdef ['=' test])* [',' '**' tfpdef] | '**' tfpdef)
+typedargslist: (tfpdef ['=' test] (',' [TYPE_COMMENT] tfpdef ['=' test])* [',' [TYPE_COMMENT]
+       ['*' [tfpdef] (',' [TYPE_COMMENT] tfpdef ['=' test])* [',' [TYPE_COMMENT] '**' tfpdef] | '**' tfpdef]] [TYPE_COMMENT]
+     |  '*' [tfpdef] (',' [TYPE_COMMENT] tfpdef ['=' test])* [',' [TYPE_COMMENT] '**' tfpdef] [TYPE_COMMENT]
+     | '**' tfpdef [TYPE_COMMENT])
```

Comment:
One consequence of this definition is that the type comment is required to come after the comma, if any, that follows the argument.

This seems fine to me -- thanks to Python's graceful handling of a redundant trailing comma, there's rarely a good reason to want to put the comma on the next line, in a form like this:
```
def do_something( options=None
                , config=None
                , force=True
                ):
```
such as might appear in other languages for the sake of uniformity between lines and avoiding unnecessary diff noise.

It's not clearly specified by PEP 484, though:
"""
Sometimes you have a long list of parameters and specifying their types in a single `# type:` comment would be awkward. To this end you may list the arguments one per line and add a `# type:` comment per line. To specify the return type use the ellipsis syntax. Not every argument needs to be given a type. A line with a `# type:` comment should contain exactly one argument. The type comment for the last argument (if any) should precede the close parenthesis.
"""

So we should say this clearly someplace like the README, and also update the PEP to clarify. (And once that's done, can take this point out of the README.)